File: GO/zadanie_2.go
package main

import "fmt"

func romanToDecimal(roman string) int {
    // Словарь, содержащий соответствие римских цифр и их десятичных значений
    romanNumerals := map[byte]int{
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000,
    }

    result := 0
    for i := 0; i < len(roman); i++ {
        // Если текущее значение больше предыдущего, вычитаем удвоенное значение предыдущего из текущего
        if i > 0 && romanNumerals[roman[i]] > romanNumerals[roman[i-1]] {
            result += romanNumerals[roman[i]] - 2*romanNumerals[roman[i-1]]
        } else {
            result += romanNumerals[roman[i]]
        }
    }
    return result
}

func main() {
    // Примеры преобразования римских чисел в десятичные и вывод результата
    fmt.Println(romanToDecimal("LXVI"))  // Вывод: 66
    fmt.Println(romanToDecimal("LXXXI")) // Вывод: 81
}

File: RUST/zadanie_2.rs
fn roman_to_decimal(roman: &str) -> i32 {
    let roman_numerals: Vec<(char, i32)> = vec![
        ('I', 1),
        ('V', 5),
        ('X', 10),
        ('L', 50),
        ('C', 100),
        ('D', 500),
        ('M', 1000)
    ];

    let mut result = 0;
    let mut prev_value = 0;
    for c in roman.chars().rev() {
        let value = roman_numerals.iter().find(|&&(r, _)| r == c).unwrap().1;
        if value < prev_value {
            result -= value;
        } else {
            result += value;
        }
        prev_value = value;
    }
    result
}

fn main() {
    println!("{}", roman_to_decimal("LXVI"));  // Output: 66
    println!("{}", roman_to_decimal("LXXXI")); // Output: 81
}
